roles/osasinfra/tasks/exec_command_in_pod.yml

```diff
@@ -0,0 +1,61 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Get current date and time
+  ansible.builtin.shell:
+    cmd: date +"%Y_%m_%d-%H_%M_%S"
+  register: current_date
+
+#- name: Execute the command '{{ command }}' in the pod '{{ namespace }}/{{ pod }}'
+#  kubernetes.core.k8s_exec:
+#    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+#    namespace: "{{ namespace }}"
+#    pod: "{{ pod }}"
+#    command: "{{ command }}"
+#  register: command_result
+#  ignore_errors: True
+
+- name: Execute the command '{{ command }}' in the pod '{{ namespace }}/{{ pod }}'
+  environment:
+    PATH: "{{ cifmw_path }}"
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+  ansible.builtin.command:
+    cmd: "oc rsh -n {{ namespace }} {{ pod }} bash -c '{{ command }}'"
```

Comment:
I would loop on a zuul variable called commands defined on https://gitlab.cee.redhat.com/ci-framework/ci-framework-jobs/-/merge_requests/100/diffs#a3101b9b5f0dae18897b3fd245799c0eb0a1a060_0_20

That way, we can list the commands to run sequentially outside ci-framework.

what do you think @imatza-rh?

Reply:
I agree. We still need to figure out getting the state in case of failure/restart etc

Reply:
For now we are running the next commands:
` - oc rsh shiftstackclient git clone https://gerrithub.io/shiftstack/shiftstack-qa`
` - oc rsh shiftstackclient bash -c "cd shiftstack-qa && ansible-galaxy collection install -r requirements.yaml -f"`
` - oc rsh shiftstackclient ansible-navigator run shiftstack-qa/playbooks/ocp_testing.yaml -e @shiftstack-qa/jobs_definitions/4.15_ovnkubernetes_ipi.yaml`

Do we expect to have a wide variety of commands?
The logic can be handled inside the job definition files as well.
Let's discuss about this ok?

Reply:
I'm ok with this approach then. Let's include everything in one single file. We can play with hooks and imports on shiftstack-qa repo.